GliaWidgets/SecureConversations/SecureConversations.AcitivityIndicator.swift

```diff
@@ -0,0 +1,202 @@
+import UIKit
+
+extension CAPropertyAnimation {
+    enum Key: String {
+        var path: String {
+            return rawValue
+        }
+
+        case strokeStart = "strokeStart"
+        case strokeEnd = "strokeEnd"
+        case strokeColor = "strokeColor"
+        case rotationZ = "transform.rotation.z"
+        case scale = "transform.scale"
+    }
+
+    convenience init(key: Key) {
+        self.init(keyPath: key.path)
+    }
+}
+
+extension CGRect {
+    var center: CGPoint {
+        return CGPoint(x: midX, y: midY)
+    }
+}
+
+extension CGSize {
+    var min: CGFloat {
+        return CGFloat.minimum(width, height)
+    }
+}
+
+extension UIBezierPath {
+    convenience init(center: CGPoint, radius: CGFloat) {
+        self.init(arcCenter: center, radius: radius, startAngle: 0, endAngle: CGFloat(.pi * 2.0), clockwise: true)
+    }
+}
+
+class ActivityIndicatorView: UIView {
+    var color: UIColor = .red {
+        didSet {
+            indicator.strokeColor = color.cgColor
+        }
+    }
+
+    var lineWidth: CGFloat = 2.0 {
+        didSet {
+            indicator.lineWidth = lineWidth
+            setNeedsLayout()
+        }
+    }
+
+    private let indicator = CAShapeLayer()
+    private let animator = ActivityIndicatorAnimator()
+
+    private var isAnimating = false
+
+    convenience init() {
+        self.init(frame: .zero)
+        self.setup()
+    }
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.setup()
+    }
+
+    private func setup() {
+        indicator.strokeColor = color.cgColor
+        indicator.fillColor = nil
+        indicator.lineWidth = lineWidth
+        indicator.strokeStart = 0.0
+        indicator.strokeEnd = 0.0
+        layer.addSublayer(indicator)
+    }
+}
+
+extension ActivityIndicatorView {
+    override public var intrinsicContentSize: CGSize {
+        return CGSize(width: 24, height: 24)
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+
+        indicator.frame = bounds
+
+        let diameter = bounds.size.min - indicator.lineWidth
```

Comment:
Is it correct that there you subtract one `lineWidth` value instead of two?

Reply:
Since line overlaps by half of its width it is fine.